services/api/src/resources/environment/environment_redeploy.ts

```diff
@@ -0,0 +1,100 @@
+// This file contains the logic to determine whether an environment requires a redeploy
+
+import * as R from 'ramda';
+import { sendToLagoonLogs } from '@lagoon/commons/dist/logs/lagoon-logger';
+import { createRemoveTask, seedNamespace } from '@lagoon/commons/dist/tasks';
+import { ResolverFn } from '..';
+import { logger } from '../../loggers/logger';
+import { isPatchEmpty, query, knex } from '../../util/db';
+import { convertDateToMYSQLDateFormat } from '../../util/convertDateToMYSQLDateTimeFormat';
+import { Helpers } from './helpers';
+import { Sql } from './sql';
+import { Sql as projectSql } from '../project/sql';
+import { Helpers as projectHelpers } from '../project/helpers';
+import { Helpers as openshiftHelpers } from '../openshift/helpers';
+import { Helpers as organizationHelpers } from '../organization/helpers';
+import { getFactFilteredEnvironmentIds } from '../fact/resolvers';
+import { getUserProjectIdsFromRoleProjectIds } from '../../util/auth';
+import { RemoveData, DeployType, AuditType } from '@lagoon/commons/dist/types';
+import { AuditLog } from '../audit/types';
+
+
+export const getPendingChangesByEnvironmentId: ResolverFn = async(
+{
+    id
+},
+_,
+{ sqlClientPool, hasPermission },
+) => {
+    // Note: as it stands, the only pending changes we have now have to do
+    // with env vars, but anything can be added in the form
+    // {type:"string", details:"string"}
```

Comment:
Minor, spotted this after the initial suggestion to add date to the result.

```suggestion
    // {type:"string", details:"string", date: "string"}
```